browser/components/MarkdownPreview.js

```diff
@@ -845,7 +845,7 @@ export default class MarkdownPreview extends React.Component {
     _.forEach(
       this.refs.root.contentWindow.document.querySelectorAll('.mermaid'),
       el => {
-        mermaidRender(el, htmlTextHelper.decodeEntities(el.innerHTML), theme)
+        mermaidRender(el, htmlTextHelper.encodeEntities(el.innerHTML), theme)
```

Comment:
Your code fixed the XSS but the mermaid is unusable after your fix.
Example:
```
```mermaid
sequenceDiagram
loop every day
    Alice->>John: Hello John, how are you?
    John-->>Alice: Great!
end
```

The encodeEntities encoded the content of the mermaid code so it becomes invalid. It seems encodeEntities is not a good way to fix this bug. @Rokt33r what do you suggest for this? Maybe a write a plugin to parse mermaid directly from markdown?

Reply:
Why is this change needed? I think the XSS is also working with-out this change - tested locally with just the `_fence` change in `markdown-it-sanitize-html.js`. 
Tested with the following [gist](https://gist.github.com/rebirthwyw/9836b2445abd4c1358e3d68ff2415247).

Reply:
> Your code fixed the XSS but the mermaid is unusable after your fix.
oh, nice catch. I confirmed that too.

@AWolf81 
Even if we sanitized a content in markdown-it-sanitize-html.js, it's going to be original in MarkdownPreview.js:715
```js
    const renderedHTML = this.markdown.render(value)
    attachmentManagement.migrateAttachments(value, storagePath, noteKey)
    this.refs.root.contentWindow.document.body.innerHTML = attachmentManagement.fixLocalURLS(
      renderedHTML,
      storagePath
    )
```
I think we have to do something for mermaid only (other plugins don't have this vulnerability).